src/ray/common/ray_config_def.h

```diff
@@ -432,3 +432,13 @@ RAY_CONFIG(uint32_t, max_error_msg_size_bytes, 512 * 1024)
 
 /// If enabled, raylet will report resources only when resources are changed.
 RAY_CONFIG(bool, enable_light_weight_resource_report, true)
+
+/// The scheduler will treat these predefined resource types as unit_instance.
+/// Default predefined_unit_instance_resources is "CPU,GPU".
+/// When set it to "GPU", we will not treat CPU as unit_instance.
+RAY_CONFIG(string_type, predefined_unit_instance_resources, "")
```

Comment:
@wuisawesome any reason we shouldn't set this to just "GPU" by default?

I haven't seen any use case for unit CPUs.

Reply:
I'm not against it, hence #10818.

That being said, I think it allows pinning a task to a cpu via `ray.resource_ids()` + cpu affinity.

Reply:
@ericl @wuisawesome How about set this to just "GPU" by default? We can add "CPU" in high performance compute scenario, so that we can bind cpu cores to different worker by cpuset/taskset, it can help to  reduce cpu competition and context switch.